frontend/packages/ui/src/ProgressBar.tsx

```diff
@@ -0,0 +1,144 @@
+import React from 'react'
+import { useState, useEffect } from 'react'
+import { BarChart, Bar, XAxis, YAxis, Tooltip, ResponsiveContainer } from 'recharts'
+import { Colors } from './styles/colors'
+import styled from 'styled-components'
+
+interface Props {
+  data: { 
+    name?: string;
+    size?: number; 
+    color?: string;
+    percentage?: number;
+    kind?: string;
+    status?: string; }[]
+  onClick: () => void
+  legend?: boolean
+}
+
+const HEIGHT = 20
+
+export const ProgressBar = ({
+    legend = false,
+    ...props
+  }: Props) => {
+
+  const [barData, setBarData] = useState(null)
+  const [barColors, setBarColors] = useState(null)
+
+  const preparePercentageChartValues = (name: 'kind' | 'status') => {
+    const data = {}
+    const colors = {}
+    props.data.map((value, i) => {
+      data[value[name]] = value.percentage
+      colors[value[name]] = Colors.progressBarColors[i%10]
+    })
+    setBarData(data)
+    setBarColors(colors)
+  }
+
+  const prepareNonPercentageChartValues = () => {
+    const data = {}
+    const colors = {}
+    props.data.map(value => {
+      data[value.name] = value.size
+      colors[value.name] = value.color
+    })
+    setBarData(data)
+    setBarColors(colors)
+  }
+
+  const prepareBarChartValues = () => 
+    props.data[0].percentage
+    ? preparePercentageChartValues(props.data[0].kind ? 'kind' : 'status')
+    : prepareNonPercentageChartValues()
+
+  useEffect(prepareBarChartValues, [props.data])
+
+  
+  const renderBars = () => {
+    const isLastBar = Object.keys(barColors).length -1
+    return Object.keys(barColors).map((bar, i) =>
+      <Bar 
+        isAnimationActive={false}
+        onClick={props.onClick}
+        key={bar}
+        fill={barColors[bar]}
+        dataKey={bar}
+        stackId={'a'}
+        radius={i === 0 && [20, 0, 0, 20] || i === isLastBar && [0, 20, 20, 0] || 0}
+      />
+    )
+  }
+
+  const renderLegend = () => {
+    return Object.keys(barData).map(bar => {
+      return (
+        <LegendPiece>
+          <LegendCircle color={barColors[bar]}/>
+          { bar }: <LegendValue>{parseFloat(barData[bar].toFixed(2))}%</LegendValue>
+        </LegendPiece>
+      )
+    })
+  }
+
+  const renderBarChart = () => {
```

Comment:
No need for a function here we can have the same code without it.

Reply:
We need it because you wanted me to use this:
`if (!barData) return null`
and in order to use it I need a function